Множественые изменения

LilKirill00 opened the pull request:
- добавлена возможность редактировать различные сообщения об ошибках пользователю через error_messages (см. Readme)
- исправлены неправильно передающиеся параметры в exec_button
- реализован параметр offer_options для save_to_var позволяющий пользователю предложить свой вариант для ответа
- исправлены мелкие ошибки
- рефактор некоторых моментов кода
- ! перемещение файлов в правильную дерикторию !

LilKirill00 committed:
refactor: добавлена возможность заменить сообщение об ошибке в процессе работы кнопки

LilKirill00 committed:
refactor: добавлено больше кастомизации ошибок

добавлен новый раздел error_messages в котором можно настроить большую
часть текстов ошибок

перенесено error_message в error_messages под новым именем
command_unknown

LilKirill00 committed:
feature: Реализована возможность предложить пользователю выбор из своих вариантов

добавлен в save_to_var параметр offer_options в котором можно
предложить варианты ответов пользователю

LilKirill00 committed:
fix: исправлена неправильная передача аргументов в exec_button

теперь переменые из шаблонов правильно передают пробелы и кавычки

LilKirill00 committed:
refactor: упрощение вызова SendAnswer

cnf.FilesDir теперь получается внутри функции а не передается

LilKirill00 committed:
fix: исправлена неправельная проверка наличия шаблона в fillTemplateWithInfo

LilKirill00 committed:
refactor: добавление коментариев и упощение функции getFileInfo

LilKirill00 committed:
fix: исправлено неправильное имя файлов при отправке

при использование следующего куска меню:
answer:
  - file: 'scripts/example.sh'
  - file: 'start.sh'

пользователь раньше получил бы 2 файла с именами "scripts/example.sh" и
"start.sh"

а сейчас получит нормальные имя без пути в виде "example.sh" и
"start.sh"

LilKirill00 committed:
fix: исправлено что при ошибке отправки файла не отправляется клавиатура

если файл шел последним в answer то при ошибке не отправлялась клавиатура

теперь выводится сообщение что произошла ошибка отправки файла и сама
клавиатура

LilKirill00 committed:
refactor: вынесена ошибка отправки файлов в ErrorMessages

можно отредактировать сообщение об ошибке отправки файла

LilKirill00 committed:
refactor: вынесены кнопки для тригера по do_button в отдельную функцию

LilKirill00 committed:
refactor: добавлено упущеное сообщение об ошибке отправки файла

LilKirill00 committed:
refactor: вынесен поиск qna в отдельную функцию

LilKirill00 committed:
refactor: вынесено в отдельные функции отправка chat и file

LilKirill00 committed:
refactor: убран не использующийся параметр в SendAnswerMenu

LilKirill00 committed:
style: добавление коментариев

LilKirill00 committed:
refactor: добавлены break чтобы не проверять лишние данные

LilKirill00 committed:
refactor: применение getGoToIfClickedBackBtn на месте дубликата

LilKirill00 committed:
fix: исправлено что do_button в menu требует текст кнопки

LilKirill00 committed:
fix: исправлен неправильный подсчет модификаторов кнопки

и исправлено невозможность дать текст по умолчанию для SaveToVar и TicketButton

LilKirill00 committed:
refactor: перенос заглушки

перенес тк defaultCreateTicketMenuBtnCnf нельзя редактировать в конфиге
и соответствено заглушку можно поставить по умолчанию

LilKirill00 committed:
refactor: исправлена заглушка для do_button в save_to_var

LilKirill00 committed:
refactor: заменен костыль на нормальный тип данных

LilKirill00 committed:
refactor: объединение 1-2 шага создания заявки с одинаковым кодом

LilKirill00 committed:
refactor: добавлена возможность использовать шаблон в SaveToVar.OfferOptions

LilKirill00 committed:
refactor: обьединение 3-5 шагов создания заявки

LilKirill00 committed:
refactor!: перемещение файлов в правильную дерикторию

polyn0m commented on bot/bot.go:
```diff
@@ -99,8 +100,13 @@ func fillTemplateWithInfo(c *gin.Context, msg *messages.Message, text string) (r
 func getFileNames(root string) map[string]bool {
 	files := make(map[string]bool)
+
+	// строим абсолютный путь к переданной директории
```
Такие комментарии излишние, загрязняют код не неся никакого смысла. Из кода и так понятно что строится абсолютный путь.  Лучше комментировать бизнес логику не ясную из кода

polyn0m commented on bot/bot.go:
```diff
@@ -99,8 +100,13 @@ func fillTemplateWithInfo(c *gin.Context, msg *messages.Message, text string) (r
+	// строим абсолютный путь к переданной директории
 	root, _ = filepath.Abs(root)
+
+	// обходим все файлы и папки
```
тоже самое

polyn0m commented on bot/bot.go:
```diff
@@ -109,59 +115,85 @@ func getFileNames(root string) map[string]bool {
 	return files
 }
 
+// проверить является ли файл изображением
```
тоже самое, можно дать название функции более понятное ну и далее в файла тоже есть, если комент очевиден с кодом, то он не нужен, код сам о себе все говорит

LilKirill00 committed:
style: убраны лишние коментарии

LilKirill00 committed:
fix: исправлено что при отправке файла может вызваться кнопка

чтобы повторить:
1) сделать кнопку в меню у которой не будет id (убедитесь что настройка
по умолчанию тоже без id)
2) попасть в меню с этой кнопкой
3) отправить файл в этом меню
в результате запустится кнопка

причина: при отправке файла text идет пустым и он сравнивается с пустым
id

LilKirill00 committed:
refactor: прошелся через linter

добавление _ на местах где ошибку не проверяем
заменен "io/ioutil" на "io"
правка имен переменых
замена неиспользующихся переменых на _

+ убран finalSend после Send, тк пытаться отправить сообщение об ошибке является бессмысленым, поскольку операцию назад была попытка отправить сообщение

LilKirill00 committed:
refactor: исправлено что не используется копия контекста

LilKirill00 committed:
refactor: вынесение в константы ответы us

LilKirill00 committed:
refactor: вынесение название этапа FINAL в константу

LilKirill00 committed:
refactor: замена перебора чисел в цикле на range

LilKirill00 committed:
refactor: улучшено понимание что не так с кнопкой

изменен вид кнопки которая выводится в ошибке
добавлено уточнение что за ошибка для ticket и savetovar

LilKirill00 committed:
docs: Добавлено объяснение по back_button
